bug: alphabetical order of resolving users when fixing duplicate issues

Thanks for the latests updates you released. 
I've found some more information about the possible bug regarding conflicts when updating resolved users. 

I am using v1.3.0 (docker) and Keycloak 9.0.2

I have a users.json like this:

```
{
  "realm": "my-realm",
  "users": [
    {
      "username": "abc@company.fi",
      "email": "abc.stakeholder@company.fi",
      "enabled": true,
      "firstName": "Abc",
      "lastName": "Stakeholder",
      "credentials": [
        {
          "type": "password",
          "value": "password"
        }
      ]
    },
    {
      "username": "company",
      "email": "company.stakeholder@company.fi",
      "enabled": true,
      "firstName": "Company",
      "lastName": "Stakeholder",
      "credentials": [
        {
          "type": "password",
          "value": "password"
        }
      ]
    }
  ]
}
```
Guess: What happens when you use keycloak search for user `company`, it will get list of these users in alphabetical order. 

In this case, I already had the `company` user and added a `abc@company.fi`
So mr. abc gets created, but when trying to update `company`, it's HTTP 409 conflict.

Is it possible that your new filter operation still returns the full list, gets the first item which is wrong in this case somehow?

stack trace:

```
2020-04-01 10:52:50.347  INFO 27 --- [           main] de.adorsys.keycloak.config.Application   : Starting Application v1.3.0_2020-03-27T20:40:28Z on a09b077849de with PID 27 (/opt/keycloak-config-cli.jar started by root in /)
... (omitted) ...
 2020-04-01 10:52:53.806 DEBUG 27 --- [           main] d.a.k.config.service.UserImportService   : Create user 'abc@company.fi' in realm 'company-application'
 2020-04-01 10:52:53.949 DEBUG 27 --- [           main] d.a.k.config.service.UserImportService   : Remove realm-level roles [uma_authorization,offline_access] from user 'abc@company.fi' in realm 'company-application'
 2020-04-01 10:52:54.019 DEBUG 27 --- [           main] d.a.k.config.service.UserImportService   : Update user 'company' in realm 'company-application'
 2020-04-01 10:52:54.138  INFO 27 --- [           main] ConditionEvaluationReportLoggingListener :

 Error starting ApplicationContext. To display the conditions report re-run your application with 'debug' enabled.
 2020-04-01 10:52:54.147 ERROR 27 --- [           main] o.s.boot.SpringApplication               : Application run failed

 java.lang.IllegalStateException: Failed to execute CommandLineRunner
 	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:787) ~[spring-boot-2.2.5.RELEASE.jar!/:2.2.5.RELEASE]
 	at org.springframework.boot.SpringApplication.callRunners(SpringApplication.java:768) ~[spring-boot-2.2.5.RELEASE.jar!/:2.2.5.RELEASE]
 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:322) ~[spring-boot-2.2.5.RELEASE.jar!/:2.2.5.RELEASE]
 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1226) ~[spring-boot-2.2.5.RELEASE.jar!/:2.2.5.RELEASE]
 	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215) ~[spring-boot-2.2.5.RELEASE.jar!/:2.2.5.RELEASE]
 	at de.adorsys.keycloak.config.Application.main(Application.java:47) ~[classes!/:1.3.0_2020-03-27T20:40:28Z]
 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[na:na]
 	at java.base/java.lang.reflect.Method.invoke(Unknown Source) ~[na:na]
 	at org.springframework.boot.loader.MainMethodRunner.run(MainMethodRunner.java:48) ~[keycloak-config-cli.jar:1.3.0_2020-03-27T20:40:28Z]
 	at org.springframework.boot.loader.Launcher.launch(Launcher.java:87) ~[keycloak-config-cli.jar:1.3.0_2020-03-27T20:40:28Z]
 	at org.springframework.boot.loader.Launcher.launch(Launcher.java:51) ~[keycloak-config-cli.jar:1.3.0_2020-03-27T20:40:28Z]
 	at org.springframework.boot.loader.JarLauncher.main(JarLauncher.java:52) ~[keycloak-config-cli.jar:1.3.0_2020-03-27T20:40:28Z]
 Caused by: javax.ws.rs.ClientErrorException: HTTP 409 Conflict
 	at org.jboss.resteasy.client.jaxrs.internal.ClientInvocation.handleErrorStatus(ClientInvocation.java:241) ~[resteasy-client-3.9.1.Final.jar!/:3.9.1.Final]
 	at org.jboss.resteasy.client.jaxrs.internal.proxy.extractors.DefaultEntityExtractorFactory$3.extractEntity(DefaultEntityExtractorFactory.java:50) ~[resteasy-client-3.9.1.Final.jar!/:3.9.1.Final]
 	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invokeSync(ClientInvoker.java:151) ~[resteasy-client-3.9.1.Final.jar!/:3.9.1.Final]
 	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientInvoker.invoke(ClientInvoker.java:112) ~[resteasy-client-3.9.1.Final.jar!/:3.9.1.Final]
 	at org.jboss.resteasy.client.jaxrs.internal.proxy.ClientProxy.invoke(ClientProxy.java:76) ~[resteasy-client-3.9.1.Final.jar!/:3.9.1.Final]
 	at com.sun.proxy.$Proxy93.update(Unknown Source) ~[na:na]
 	at de.adorsys.keycloak.config.repository.UserRepository.updateUser(UserRepository.java:86) ~[classes!/:1.3.0_2020-03-27T20:40:28Z]
 	at de.adorsys.keycloak.config.service.UserImportService$UserImport.updateUser(UserImportService.java:101) ~[classes!/:1.3.0_2020-03-27T20:40:28Z]
 	at de.adorsys.keycloak.config.service.UserImportService$UserImport.importUser(UserImportService.java:86) ~[classes!/:1.3.0_2020-03-27T20:40:28Z]
 	at de.adorsys.keycloak.config.service.UserImportService.importUser(UserImportService.java:68) ~[classes!/:1.3.0_2020-03-27T20:40:28Z]
 	at de.adorsys.keycloak.config.service.UserImportService.doImport(UserImportService.java:61) ~[classes!/:1.3.0_2020-03-27T20:40:28Z]
 	at de.adorsys.keycloak.config.service.RealmImportService.updateRealm(RealmImportService.java:176) ~[classes!/:1.3.0_2020-03-27T20:40:28Z]
 	at de.adorsys.keycloak.config.service.RealmImportService.updateRealmIfNecessary(RealmImportService.java:157) ~[classes!/:1.3.0_2020-03-27T20:40:28Z]
 	at de.adorsys.keycloak.config.service.RealmImportService.doImport(RealmImportService.java:132) ~[classes!/:1.3.0_2020-03-27T20:40:28Z]
 	at de.adorsys.keycloak.config.Application.run(Application.java:57) ~[classes!/:1.3.0_2020-03-27T20:40:28Z]
 	at org.springframework.boot.SpringApplication.callRunner(SpringApplication.java:784) ~[spring-boot-2.2.5.RELEASE.jar!/:2.2.5.RELEASE]
 	... 13 common frames omitted

company-application-keycloak-config exited with code 1
```

--- a/src/main/java/de/adorsys/keycloak/config/repository/UserRepository.java
+++ b/src/main/java/de/adorsys/keycloak/config/repository/UserRepository.java
@@ -44,14 +44,14 @@ public class UserRepository {
 
     public Optional<UserRepresentation> tryToFindUser(String realm, String username) {
         Optional<UserRepresentation> maybeUser;
-        List<UserRepresentation> foundUsers = realmRepository.loadRealm(realm).users().search(username);
-        List<UserRepresentation> filteredUsers = foundUsers.stream()
-                .filter(u -> u.getUsername().equalsIgnoreCase(username)).collect(Collectors.toList());
 
-        if (filteredUsers.isEmpty()) {
+        try {
+            UserRepresentation user = findUser(realm, username);
+
+            maybeUser = Optional.of(user);
+        }
+        catch(KeycloakRepositoryException e) {
             maybeUser = Optional.empty();
-        } else {
-            maybeUser = Optional.of(filteredUsers.get(0));
         }
 
         return maybeUser;
@@ -64,12 +64,14 @@ public class UserRepository {
 
     public UserRepresentation findUser(String realm, String username) throws KeycloakRepositoryException {
         List<UserRepresentation> foundUsers = realmRepository.loadRealm(realm).users().search(username);
+        List<UserRepresentation> filteredUsers = foundUsers.stream()
+                .filter(u -> u.getUsername().equalsIgnoreCase(username)).collect(Collectors.toList());
 
-        if (foundUsers.isEmpty()) {
+        if (filteredUsers.isEmpty()) {
             throw new KeycloakRepositoryException("Cannot find user '" + username + "' in realm '" + realm + "'");
         }
 
-        return foundUsers.get(0);
+        return filteredUsers.get(0);
     }
 
     public void create(String realm, UserRepresentation userToCreate) throws KeycloakRepositoryException {

--- a/src/test/java/de/adorsys/keycloak/config/ImportUsersIT.java
+++ b/src/test/java/de/adorsys/keycloak/config/ImportUsersIT.java
@@ -206,4 +206,22 @@ public class ImportUsersIT extends AbstractImportTest {
         assertThat(updatedUser2.getLastName(), is("lastName2"));
         assertThat(updatedUser2.getFirstName(), is("firstName2"));
     }
+
+    /**
+     * https://github.com/adorsys/keycloak-config-cli/issues/68
+     */
+
+    @Test
+    @Order(5)
+    public void coverGitHubIssue68() {
+        // Create Users
+        doImport("5_1_issue_gh_68.json");
+        // Update Users
+        doImport("5_2_issue_gh_68.json");
+
+        RealmResource realmResource = keycloakProvider.get().realm(REALM_NAME);
+        final RealmRepresentation createdRealm = realmResource.toRepresentation();
+        assertThat(createdRealm.getRealm(), is(REALM_NAME));
+        assertThat(createdRealm.isEnabled(), is(true));
+    }
 }

--- a/src/test/resources/import-files/users/5_1_issue_gh_68.json
+++ b/src/test/resources/import-files/users/5_1_issue_gh_68.json
@@ -1,0 +1,19 @@
+{
+  "enabled": true,
+  "realm": "realmWithUsers",
+  "users": [
+    {
+      "username": "company",
+      "email": "company.stakeholder@company.fi",
+      "enabled": false,
+      "firstName": "Company",
+      "lastName": "Stakeholder",
+      "credentials": [
+        {
+          "type": "password",
+          "value": "password"
+        }
+      ]
+    }
+  ]
+}

--- a/src/test/resources/import-files/users/5_2_issue_gh_68.json
+++ b/src/test/resources/import-files/users/5_2_issue_gh_68.json
@@ -1,0 +1,32 @@
+{
+  "enabled": true,
+  "realm": "realmWithUsers",
+  "users": [
+    {
+      "username": "abc@company.fi",
+      "email": "abc.stakeholder@company.fi",
+      "enabled": true,
+      "firstName": "Abc",
+      "lastName": "Stakeholder",
+      "credentials": [
+        {
+          "type": "password",
+          "value": "password"
+        }
+      ]
+    },
+    {
+      "username": "company",
+      "email": "company.stakeholder@company.fi",
+      "enabled": true,
+      "firstName": "Company",
+      "lastName": "Stakeholder",
+      "credentials": [
+        {
+          "type": "password",
+          "value": "password"
+        }
+      ]
+    }
+  ]
+}

